### PR TITLE
Sprints/summerofsprinty9

### PR DESCRIPTION
**Pull request by GabrielFG**

### CODE REVIEW

File: src/Compliance.Solution/Solution/Entities/opc_informationcentre/RibbonDiff.xml [hunk 1]
```diff
@@ -1,34 +1,29 @@
...
+    <HideCustomAction HideActionId="cms.Mscrm.HomepageGrid.opc_informationcentre.Import.Hide" Location="Mscrm.HomepageGrid.opc_informationcentre.Import" />
+    <HideCustomAction HideActionId="cms.Mscrm.HomepageGrid.opc_informationcentre.RunScript.Hide" Location="Mscrm.HomepageGrid.opc_informationcentre.RunScript" />
+    <CustomAction Id="opc.cms.ATIP.GeneratePDF.CustomAction" Location="Mscrm.Form.opc_informationcentre.MainTab.Save.Controls._children" Sequence="75">
       <CommandUIDefinition>
         <Button Command="cms.ATIP.GeneratePDF" Id="cms.ATIP.GeneratePDF" Image32by32="/_imgs/ribbon/runreport32.png" Image16by16="/_imgs/ribbon/RunReport_16.png" LabelText="$LocLabels:cms.ATIP.GeneratePDF.LabelText" Sequence="75" TemplateAlias="o2" />
       </CommandUIDefinition>
     </CustomAction>
-    <CustomAction Id="cms.Form.opc_informationcentre.ChangeOwner.CustomAction" Location="Mscrm.Form.opc_informationcentre.MainTab.Save.Controls._children" Sequence="75">
+    <CustomAction Id="opc.cms.Form.opc_informationcentre.ChangeOwner.CustomAction" Location="Mscrm.Form.opc_informationcentre.MainTab.Save.Controls._children" Sequence="75">
```
Comment (JonathanThomassin) [inline src/Compliance.Solution/Solution/Entities/opc_informationcentre/RibbonDiff.xml:16]: Should the ".cms" be removed?

File: src/Compliance.Solution/Solution/Entities/opc_informationcentre/Entity.xml [hunk 2]
```diff
@@ -1002,19 +1002,53 @@
           <LookupStyle>single</LookupStyle>
           <LookupTypes />
           <displaynames>
-            <displayname description="Information Officer" languagecode="1033" />
+            <displayname description="ZDEL_Information Officer" languagecode="1033" />
```
Comment (GabrielFG) [inline src/Compliance.Solution/Solution/Entities/opc_informationcentre/Entity.xml:1005]: Can we delete this now or do we still need it?

File: src/Compliance.Plugins.Tests/ComplaintSummaryPluginTests.cs [hunk 3]
```diff
@@ -360,9 +360,144 @@ public void summary_should_be_overwritten()
...
+            var context = new XrmFakedContext();
+            var pluginContext = context.GetDefaultPluginContext();
+            var formattedValues = new KeyValuePair<string, string>[]
+{
+                    new KeyValuePair<string, string>("opc_respondentandcomplainantrelationship", "Tenant"),
+                    new KeyValuePair<string, string>("opc_complainantperceivedpriorities","Acknowledgement; Apology; Court"),
+                    new KeyValuePair<string, string>("opc_complaintdisposition","Settled"),
+};
+            opc_dispositionreason dispostitionReason;
```
Comment (JonathanThomassin) [inline src/Compliance.Plugins.Tests/ComplaintSummaryPluginTests.cs:376]: Typo ;)

File: src/Compliance.Solution/Solution/Other/Customizations.xml [hunk 4]
```diff
@@ -36,6 +36,20 @@
...
+      <statecode>0</statecode>
+      <statuscode>1</statuscode>
+    </connectionreference>
+    <connectionreference connectionreferencelogicalname="opc_sharedmicrosoftforms_77983">
+      <connectionreferencedisplayname>Microsoft Forms</connectionreferencedisplayname>
+      <connectorid>/providers/Microsoft.PowerApps/apis/shared_microsoftforms</connectorid>
+      <iscustomizable>1</iscustomizable>
+      <statecode>0</statecode>
+      <statuscode>1</statuscode>
+    </connectionreference>
```
Comment (GabrielFG) [inline src/Compliance.Solution/Solution/Other/Customizations.xml:52]: Remove...

File: src/Compliance.WebResources/compliance/forms/ComplaintMainForm.test.ts [hunk 5]
```diff
@@ -157,6 +157,17 @@ describe("Complaint", () => {
...
+
+            it("it should load into the general tab", () => {
+                // Arrange
+                formContext.ui.setFormType(Xrm.FormType.Create);
+
+                // Act
+                form.initializeComponents(mockContext);
+
+                // Assert
+                formContext.ui.tabs.get("tab_summary").getVisible().should.equal(false);
```
Comment (StephBrouil) [inline src/Compliance.WebResources/compliance/forms/ComplaintMainForm.test.ts:169]: We should probably add an assert for tab_general visibility.

File: src/Compliance.WebResources/compliance/forms/ComplaintMainForm.ts [hunk 6]
```diff
@@ -72,6 +72,25 @@ export namespace Complaint.Forms {
...
+            } else {
+                formContext.ui.tabs.get("tab_summary").setVisible(true);
+                const currentStage = formContext.data.process.getActiveStage().getName().toLowerCase();
+
+                switch (currentStage) {
+                    case "acceptance":
+                    case "review":
+                    case "close":
+                        formContext.ui.tabs.get("tab_general").setFocus();
+                        break;
```
Comment (StephBrouil) [inline src/Compliance.WebResources/compliance/forms/ComplaintMainForm.ts:91]: This seems like a side effect that doesn't belong here?

File: src/Compliance.Solution/Solution/Entities/opc_informationcentre/Entity.xml (hunk 2 above)
Comment (GabrielFG) [reply src/Compliance.Solution/Solution/Entities/opc_informationcentre/Entity.xml:1005]: We'll keep it for now.

File: src/Compliance.WebResources/compliance/forms/ComplaintMainForm.test.ts (hunk 5 above)
Comment (GabrielFG) [reply src/Compliance.WebResources/compliance/forms/ComplaintMainForm.test.ts:169]: @nlelicanin

File: src/Compliance.WebResources/compliance/forms/ComplaintMainForm.ts (hunk 6 above)
Comment (GabrielFG) [reply src/Compliance.WebResources/compliance/forms/ComplaintMainForm.ts:91]: @nlelicanin

**Review by joshcampbell191 — APPROVED** [review]

**Pull request merged**